File: 2018/12-Dec/12-22-multiple-inheritance/php/ChristmasTree.php
<?php
/**
 * Create a trait for a Christmas tree
 * User: Andrew Jarombek
 * Date: 12/17/2018
 */

trait ChristmasTree
{

    public $holidayTree = true;

    // PHP Fatal error:  Trait method type has not been applied, because there are collisions with other trait methods
    // Methods from traits used in a class must be unique.  If multiple traits have methods of the same name,
    // a conflict occurs.

    //public function type() {return "Christmas";}
}

File: 2018/12-Dec/12-22-multiple-inheritance/csharp/ChristmasTree.cs
/**
 * Concrete class for a Christmas tree
 * Author: Andrew Jarombek
 * Date: 1/14/2019
 */

namespace Multiple_Inheritance
{
    public sealed class ChristmasTree : Tree
    {
        /// <inheritdoc cref="Tree.Type"/>
        public string Type() => "Christmas";

        /// <inheritdoc cref="Tree.Height"/>
        public int Height() => throw new System.NotImplementedException();
    }
}